crates/gitbutler-branch-actions/src/base.rs

```diff
@@ -550,6 +536,34 @@ pub(crate) fn update_base_branch(
     Ok(unapplied_branch_names)
 }
 
+/// try to update the target branch
+/// this means that we need to:
+/// - determine if what the target branch is now pointing to is mergeable with our current working directory,
+/// - merge the target branch into our current working directory
+/// - update the target sha
+/// - return all conflicting references that were unapplied to avoid the conflict
+pub(crate) fn update_base_branch(
+    ctx: &CommandContext,
+    perm: &mut WorktreeWritePermission,
+) -> anyhow::Result<Vec<ReferenceName>> {
+    ctx.assure_resolved()?;
+
+    // look up the target and see if there is a new oid
+    let target = default_target(&ctx.project().gb_dir())?;
+    let repo = ctx.repository();
+    let target_branch = repo
+        .find_branch_by_refname(&target.branch.clone().into())
+        .context(format!("failed to find branch {}", target.branch))?;
```

Comment:
Just as a not: Whenever something is non-static thanks to formatting, using `context()` will always pay for the `format!()` even though it's not used.
Ideally, get used to delay the work for when it's actually needed using `with_context(|| format!(…))`.
Of course, it's fair to optimise for readability which is when one would rather do it like this, and performance doesn't matter outside of hot-loops and when it's measured.

Reply:
I see, thanks for the advice!
Out of curiosity: Is there any downside to only using `with_context` aside from the small readability penalty?

Reply:
I don't think so, even though admittedly I'd use it if the string is static.
And again, I'd keep it this way, i.e. with `context()`, as I'd optimize for readability most of the time. When I write this I will always write `.with_context(|| …)` probably simply because… I just feel I have to write more efficient code if the trade-off seems acceptable. But that's a me *problem* I think 😅.

Reply:
Fair enough :)

Yeah, to me it makes sense to get as many performance gains if the price is that small, really.
Will keep it in mind!

Reply:
I think this is my point - there will be no gain by doing this (`with_context()`), nobody will ever notice. That's why what's done here, the simple thing, is probably best. It's just that I can't do it because of being a bit inflexible.

Reply:
😂